ipn/ipnlocal/extension_host.go

```diff
@@ -635,6 +671,31 @@ func (h *ExtensionHost) enqueueBackendOperation(f func(Backend)) {
 	}
 }
 
+func (h *ExtensionHost) RegisterOptionSetter(typ reflect.Type, f func(any) error) {
+	if h == nil {
+		// Because it's required by TestNilExtensionHostMethodCall.
+		// TODO(bradfitz): I'd rather panic here.
+		return
+	}
+	if _, ok := h.optsSetter[typ]; ok {
+		panic(fmt.Sprintf("option setter for %T already registered", typ))
```

Comment:
I think this should be
```suggestion
		panic(fmt.Sprintf("option setter for %v already registered", typ))
```
as `typ` is a `reflect.Type` rather than a value of that type.